{
	"document": {
		"aggregate_severity": {
			"namespace": "https://www.suse.com/support/security/rating/",
			"text": "important"
		},
		"category": "csaf_vex",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright 2025 SUSE LLC. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "SUSE CVE-2021-29552",
				"title": "Title"
			},
			{
				"category": "description",
				"text": "TensorFlow is an end-to-end open source platform for machine learning. An attacker can cause a denial of service by controlling the values of `num_segments` tensor argument for `UnsortedSegmentJoin`. This is because the implementation(https://github.com/tensorflow/tensorflow/blob/a2a607db15c7cd01d754d37e5448d72a13491bdb/tensorflow/core/kernels/unsorted_segment_join_op.cc#L92-L93) assumes that the `num_segments` tensor is a valid scalar. Since the tensor is empty the `CHECK` involved in `.scalar<T>()()` that checks that the number of elements is exactly 1 will be invalidated and this would result in process termination. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also affected and still in supported range.",
				"title": "Description of the CVE"
			},
			{
				"category": "legal_disclaimer",
				"text": "CSAF 2.0 data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
				"title": "Terms of use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://www.suse.com/support/security/contact/",
			"name": "SUSE Product Security Team",
			"namespace": "https://www.suse.com/"
		},
		"references": [
			{
				"category": "external",
				"summary": "CVE-2021-29552",
				"url": "https://www.suse.com/security/cve/CVE-2021-29552"
			},
			{
				"category": "external",
				"summary": "SUSE Security Ratings",
				"url": "https://www.suse.com/support/security/rating/"
			},
			{
				"category": "external",
				"summary": "SUSE Bug 1186108 for CVE-2021-29552",
				"url": "https://bugzilla.suse.com/1186108"
			}
		],
		"title": "SUSE CVE CVE-2021-29552",
		"tracking": {
			"current_release_date": "2025-01-01T04:35:07Z",
			"generator": {
				"date": "2023-02-15T03:43:15Z",
				"engine": {
					"name": "cve-database.git:bin/generate-csaf-vex.pl",
					"version": "1"
				}
			},
			"id": "CVE-2021-29552",
			"initial_release_date": "2023-02-15T03:43:15Z",
			"revision_history": [
				{
					"date": "2023-02-15T03:43:15Z",
					"number": "2",
					"summary": "Current version"
				},
				{
					"date": "2025-01-01T04:35:07Z",
					"number": "3",
					"summary": "Current version"
				}
			],
			"status": "interim",
			"version": "3"
		}
	},
	"product_tree": {}
}
